fe/fe-core/src/main/java/org/apache/doris/analysis/SysVariableDesc.java

```diff
@@ -189,6 +189,15 @@ public boolean equals(Object obj) {
         if (!setType.equals(((SysVariableDesc) obj).getSetType())) {
             return false;
         }
-        return literalExpr.equals(((SysVariableDesc) obj).getLiteralExpr());
+
+        if (literalExpr != null) {
+            return literalExpr.equals(((SysVariableDesc) obj).getLiteralExpr());
+        } else {
+            if (((SysVariableDesc) obj).getLiteralExpr() == null) {
+                return true;
+            } else {
+                return false;
+            }
+        }
```

Comment:
You can just type the following code to achieve the same effect.

```Java
Objects.equals(literalExpr, ((SysVariableDesc) obj).getLiteralExpr());
```

Besides, don't forget `import java.util.Objects;`.

Reply:
nice, done